lib/cloud-run-services.js

```diff
@@ -70,4 +73,73 @@ export async function getService(projectId, location, serviceId) {
     }
     throw error; // Re-throw other errors
   }
-}
\ No newline at end of file
+}
+
+/**
+ * Fetches logs for a specific Cloud Run service.
+ * @param {string} projectId - The Google Cloud project ID.
+ * @param {string} location - The Google Cloud location (e.g., 'europe-west1').
+ * @param {string} serviceId - The ID of the Cloud Run service.
+ * @returns {Promise<Array<object>>} - A promise that resolves to an array of log entry objects.
+ */
+export async function getServiceLogs(projectId, location, serviceId) {
+  if (!loggingClient) {
+    const { Logging } = await import('@google-cloud/logging');
+    loggingClient = new Logging({ projectId });
+  }
+
+  try {
+    const LOG_SEVERITY = 'DEFAULT'; // e.g., 'DEFAULT', 'INFO', 'WARNING', 'ERROR'
+    const PAGE_SIZE = 50;       // Number of log entries to retrieve
+
+    const filter = `resource.type="cloud_run_revision"
+                    resource.labels.service_name="${serviceId}"
+                    resource.labels.location="${location}"
+                    severity>=${LOG_SEVERITY}`;
+
+    console.log(`Getting details for Cloud Run service ${serviceId} in project ${projectId}, location ${location}...`);
+
+    const [entries] = await loggingClient.getEntries({
+      filter: filter,
+      orderBy: 'timestamp desc', // Get the latest logs first
+      pageSize: PAGE_SIZE,
+    });
+
+    // return entries; // Original return
+    const formattedLogLines = entries.map(entry => formatLogEntry(entry)).join('\n');
+    return formattedLogLines
+  } catch (error) {
+    console.error(`Error fetching logs for Cloud Run service ${serviceId}:`, error);
+    throw error;
+  }
+}
+
+/**
+ * Formats a single log entry for display.
+ * @param {object} entry - A log entry object from the Cloud Logging API.
+ * @returns {string} - A formatted string representation of the log entry.
+ */
+function formatLogEntry(entry) {
+  const timestampStr = entry.metadata.timestamp.toISOString() || 'N/A';
+  const severity = entry.metadata.severity || 'N/A';
+  let responseData = ''
+  if (entry.metadata.httpRequest) {
+    const responseMethod = entry.metadata.httpRequest.requestMethod;
+    const responseCode = entry.metadata.httpRequest.status;
+    const requestUrl = entry.metadata.httpRequest.requestUrl;
+    const responseSize = entry.metadata.httpRequest.responseSize;
+    responseData = `HTTP Request: ${responseMethod} StatusCode: ${responseCode} ResponseSize: ${responseSize} Byte - ${requestUrl}`
+  }
+
+  let data = ''
+  if (entry.data && entry.data.value) {
+    const protopath = protofiles.getProtoPath('../google/cloud/audit/audit_log.proto')
```

Comment:
Do we really need proto to parse the LogEntry?
it's a standard JSON.

Reply:
the standard Log entries are standard JSON. But all the audit log LogEntires (revision updates, revision status,....) are protoBufs, as far as I understand. 

Do you know another way for the Audit logs?

Reply:
Ah I see.